### ISSUE TITLE
How to know if it works？Wav2Vec2.0 

### ISSUE THREAD

**Issue by Z-yq** [pytorch/fairseq#3109]
## ❓ Questions and Help

I try to use the code on my own data set. 

From the training logs, how can I know if it is working properly or the current performance?

**Comment by jubick1337** [comment]
Could you specify what are you trying to do?
Finetuning wav2vec2 or pretraining?

**Comment by Z-yq** [comment]
@jubick1337 Thanks for help!

I'm using it for pre-training and finetuning  now.

For pre-training:

I tried two of my own data sets:
Data-1  logs:

> [train_inner][INFO] - {"epoch": 1, "update": 0.885, "loss": "6.658", "ntokens": "948.255", "nsentences": "18.06", "prob_perplexity": "639.979", "code_perplexity": "353.274", "temp": "0.65", "loss_0": "6.658", "loss_1": "0", "loss_2": "0", "accuracy": "0.42083", "wps": "2841.1", "ups": "3", "wpb": "948.3", "bsz": "18.1", "num_updates": "225000", "lr": "0.00047568", "gnorm": "0", "loss_scale": "4096", "train_wall": "30", "wall": "86018"}
> [fairseq_cli.train][INFO] - begin validation on "valid" subset
> [valid][INFO] - {"epoch": 1, "valid_loss": "6.16", "valid_ntokens": "814.394", "valid_nsentences": "18.8167", "valid_prob_perplexity": "639.979", "valid_code_perplexity": "37.608", "valid_temp": "0.649", "valid_loss_0": "6.16", "valid_loss_1": "0", "valid_loss_2": "0", "valid_accuracy": "0.66425", "valid_wps": "19902.9", "valid_wpb": "814.4", "valid_bsz": "18.8", "valid_num_updates": "225000", "valid_best_loss": "4.028"}


Data-2  logs:

> [train_inner][INFO] - {"epoch": 1, "update": 0.212, "loss": "6.658", "ntokens": "1699.01", "nsentences": "5.135", "prob_perplexity": "639.979", "code_perplexity": "516.391", "temp": "0.506", "loss_0": "6.658", "loss_1": "0", "loss_2": "0", "accuracy": "0.00085", "wps": "6074", "ups": "3.58", "wpb": "1699", "bsz": "5.1", "num_updates": "275000", "lr": "0.00046938", "gnorm": "0", "loss_scale": "8", "train_wall": "51", "wall": "80966"}
> [fairseq_cli.train][INFO] - begin validation on "valid" subset
> [valid][INFO] - {"epoch": 1, "valid_loss": "5.053", "valid_ntokens": "1684.31", "valid_nsentences": "5.23539", "valid_prob_perplexity": "639.98", "valid_code_perplexity": "19.384", "valid_temp": "0.506", "valid_loss_0": "5.053", "valid_loss_1": "0", "valid_loss_2": "0", "valid_accuracy": "0.66461", "valid_wps": "10098.3", "valid_wpb": "1684.3", "valid_bsz": "5.2", "valid_num_updates": "275000", "valid_best_loss": "0.756"}

Their training accuracy are quite the opposite, so which values should I pay attention to to to confirm whether the training is effective。

For Finetunning:
I tried it on Data-2,and logs:

> [train_inner][INFO] - {"epoch": 1, "update": 0.908, "loss": "198.911", "ntokens": "11334.6", "nsentences": "729.6", "nll_loss": "12.804", "wps": "1003.1", "ups": "0.09", "wpb": "11334.6", "bsz": "729.6", "num_updates": "630", "lr": "1.60925e-05", "gnorm": "345.954", "loss_scale": "0.25", "train_wall": "5", "wall": "6975"}
 
Similarly, which value should I focus on?

**Comment by jubick1337** [comment]
For pretraining part look at ```valid_loss``` in ```[valid][INFO]```
For fine-tuning based on this logs validation have not passed yet but it should be similar in structure as pretraining. So it will have  ```valid_loss``` in ```[valid][INFO]``.
Basically you want validation loss to decrease as time goes on. 
I can't tell you what loss value is good because it depends on many things.
